pkg/server/server.go

```diff
@@ -620,14 +610,17 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 	}
 
 	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
-
-	checkResolver := graph.NewLocalChecker(
-		storagewrappers.NewCombinedTupleReader(s.datastore, req.ContextualTuples.GetTupleKeys()),
-		s.checkOptions...,
+	ctx = storage.ContextWithRelationshipTupleReader(ctx,
+		storagewrappers.NewBoundedConcurrencyTupleReader(
+			storagewrappers.NewCombinedTupleReader(
```

Comment:
shouldn't we invert the order of these? otherwise i think we will try to bound concurrency for the contextual tuples 🤔

Reply:
No, because the counded concurrency tuple reader just bounds concurrent calls to the storage methods. The combined reader just merges the remote storage with local in-memory contextual tuples. The outer layer should be the bounded concurrency reader. This is how it is today.

We construct the BoundedConcurrencyTupleReader around what we pass into `NewLocalChecker`, and we pass in the CombinedTupleReader into `NewLocalChecker`. See:
* https://github.com/openfga/openfga/blob/main/pkg/server/server.go#L625
* https://github.com/openfga/openfga/blob/main/internal/graph/check.go#L177

So effectively what we have is:

```
NewBoundedConcurrencyTupleReader(
    NewCombinedTupleReader(baseDatastore, req.GetContextualTuples()),
)
```

These changes mimic that behavior in a functionally equivalent manner.

Reply:
> This is how it is today.

So this is broken in main today too?

Reply:
I don't understand why it's broken. Please elaborate more with an example or test if possible. That would help direct the conversation more concretely.

The way I see it is that the inner combined tuple reader just merges in the contextual tuples, but the bounded concurrency reader limits how many invocations of `Read`, `ReadUserTuple`, `ReadUsersetTuples`, etc.. can get called concurrently. I don't see that being an issue. The combined tuple reader still has to hit the database to join the tuples in remote storage with the contextual tuples in memory. So you still want to bound that concurrency to the remote storage layer.